firmware/robot2015/src-ctrl/modules/robot-expander-inputs.hpp

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+#include <mbed.h>
+#include <rtos.h>
+
+#include <mcp23017.hpp>
+#include <io-expander.hpp>
+#include <helper-funcs.hpp>
+#include <logger.hpp>
+#include <Console.hpp>
+
+#include "pins-ctrl-2015.hpp"
+#include "task-signals.hpp"
+
+/**
+ * A DigitalInOut class meant to replicate basic functionality of the
+ * mBed DigitalOut and DigitalIn on the io-expander.
+ */
+class RobotExpanderInputs {
+public:
+    /// Other constructors for creating objects for pinouts
+    RobotExpanderInputs(MCP23017* mcp, PinName interruptPin)
+        : interruptThread_(&RobotExpanderInputs::ThreadHelper, this,
+                           osPriorityBelowNormal, 0.33 * DEFAULT_STACK_SIZE),
+          mcp23017_(mcp),
+          interrupt_(interruptPin) {
+        interrupt_.fall(this, &RobotExpanderInputs::ISR);
+        interrupt_.mode(PullUp);
+    }
+
+    uint8_t RotarySelector() {
```

Comment:
Keep in mind we also have the RotarySelector class that does the same thing as this method.  The only benefit over the other implementation is that it works with io expander pins as well as regular mbed pins.